nixos/modules/system/boot/loader/limine/limine.nix

```diff
@@ -0,0 +1,210 @@
+    };
+
+    enableEditor = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether to allow editing the boot entries before booting them.
+        It is recommended to set this to false, as it allows gaining root
+        access by passing `init=/bin/sh` as a kernel parameter.
+      '';
+    };
+
+    maxGenerations = mkOption {
+      default = null;
+      example = 50;
+      type = types.nullOr types.int;
+      description = ''
+        Maximum number of latest generations in the boot menu.
+        Useful to prevent boot partition of running out of disk space.
+        `null` means no limit i.e. all generations that were not
+        garbage collected yet.
+      '';
+    };
+
+    extraConfig = mkOption {
+      default = "";
+      type = types.lines;
+      example = literalExpression ''
+        /memtest86
+          protocol: chainload
+          path: boot():///efi/memtest86/memtest86.efi
+      '';
+      description = ''
+        A string which is appended to the end of limine.conf. The config format can be found [here](https://github.com/limine-bootloader/limine/blob/trunk/CONFIG.md).
+      '';
+    };
+
+    additionalFiles = mkOption {
+      default = { };
+      type = types.attrsOf types.path;
+      example = literalExpression ''
+        { "efi/memtest86/memtest86.efi" = "''${pkgs.memtest86-efi}/BOOTX64.efi"; }
+      '';
+      description = ''
+        A set of files to be copied to {file}`/boot`. Each attribute name denotes the
+        destination file name in {file}`/boot`, while the corresponding attribute value
+        specifies the source file.
+      '';
+    };
+
+    useStorePaths = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Use direct Nix store paths instead of copying boot files onto the boot partition. This is
+        noticeably slower but can greatly reduce the space usage on the boot partition.
+      '';
+    };
+
+    validateChecksums = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to validate file checksums before booting.
+      '';
+    };
+
+    panicOnChecksumMismatch = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether or not checksum validation failure should be a fatal
+        error at boot time.
+      '';
+    };
+
+    package = mkOption {
+      default = pkgs.limine;
+      defaultText = literalExpression "pkgs.limine";
+      type = types.package;
+      description = ''
+        Which Limine package to use for installation.
+      '';
+    };
+
+    installEfi = mkOption {
+      default = pkgs.stdenv.hostPlatform.isEfi;
+      defaultText = literalExpression "true if on EFI";
+      type = types.bool;
+      description = ''
+        Whether or not to install the limine EFI files.
+      '';
+    };
+
+    installBios = mkOption {
+      default = pkgs.stdenv.hostPlatform.isi686;
```

Comment:
BIOS on x86_64 and UEFI on i686 both exist, I think this should be `!installEfi && isx86` or similar.